Keep Managed Extensions During Installation

manuth opened the pull request:
Whenever `installExtensions` is executed, all existing managed extensions are stripped from the extensions file.
This causes the extensions file, whenever a change happens (through `applyChanges` or a manual `installExtensions` invocation from the command palette), to be cleared.

Changes made in this PR fix this issue.

daiyam commented:
@manuth The extensions in `managedExtensions` and not in `installedExtensions` are uninstalled by https://github.com/zokugun/vscode-vsix-manager/blob/3ff19d2fffc189881c85b96abf2871e7c39d3576/src/commands/install-extensions.ts#L48

So `await fse.writeJSON(extensionsFileName, installedExtensions);` is correct.

The uninstall is to remove extensions that aren't managed by the manager anymore.

manuth commented:
I feel like this might lead to unwanted behaviors.

Imagine I have an entry `muhammad-sammy.csharp` in my list of extensions to be installed by `vsix-manager`

What happens to my unserstanding according to the code is the following:

- When running `installExtensions` for the first time
  - vsix-manager will notice that `muhammad-sammy.csharp` isn't installed
  - `muhammad-sammy.csharp` will be installed and added to `installedExtensions`
  - The `installExtensions` are written to a file `extensions.json`
- When running `installExtensions` a second time (manually, through cron job or during extension activation)
  - vsix-manager will notice that `muhammad-sammy.csharp` is installed already and skip it without it being added to `installedExtensions`
  - With `muhammad-sammy.csharp` being part of the `managedExtensions` but not part of `installedExtensions`, despite the fact that I have it listed in `vsix-manager.extensions`, the extension will be uninstalled

My original issue, actually is, that when running `installExtensions` twice, it removes all existing entries from the `extensions.json` file. But taking the line you have linked in the comment before, it well might even cause that the extensions are both uninstalled and removed from the `extensions.json` file even if I have them listed in the `vsix-manager.extensions` setting.

I consider both (the uninstall and the removal from `extensions.json`) unwanted behaviors.

daiyam commented:
@manuth You are right there is an issue! I've made #6 to fix it. Can you take a look if it's good for you?

manuth commented:
Wow that was fast!
Thanks a ton, I'll give it a go asap!
